ouroboros-network/src/Ouroboros/Network/Protocol/LocalTxSubmission/Test.hs

```diff
@@ -0,0 +1,232 @@
+{-# LANGUAGE NamedFieldPuns            #-}
+{-# LANGUAGE FlexibleInstances          #-}
+{-# LANGUAGE GADTs                      #-}
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+{-# LANGUAGE RankNTypes                 #-}
+{-# LANGUAGE KindSignatures             #-}
+{-# LANGUAGE PolyKinds                  #-}
+{-# LANGUAGE DataKinds                  #-}
+
+{-# OPTIONS_GHC -Wno-orphans #-}
+
+module Ouroboros.Network.Protocol.LocalTxSubmission.Test (
+    tests
+  ) where
+
+import           Data.ByteString.Lazy (ByteString)
+
+import           Control.Monad.ST (runST)
+import           Control.Monad.IOSim
+import           Control.Monad.Class.MonadAsync (MonadAsync)
+import           Control.Monad.Class.MonadST (MonadST)
+import           Control.Monad.Class.MonadThrow (MonadCatch)
+import           Control.Tracer (nullTracer)
+
+import           Codec.Serialise (Serialise)
+import qualified Codec.Serialise as Serialise (encode, decode)
+
+import           Network.TypedProtocol.Core
+import           Network.TypedProtocol.Driver
+import           Network.TypedProtocol.Proofs
+import           Ouroboros.Network.Channel
+import           Ouroboros.Network.Codec hiding (prop_codec)
```

Comment:
In other modules we use `prop_codec_<ProtocolTag>` to avoid this clash, maybe we should rename the `prop_codec`, e.g. use a different prefix for utility functions exported from `Network.TypedProtocol.Codec` like `test_`.

Reply:
I think short names in the local module is fine. So yes if we want to avoid the clash then renaming the general `prop_codec` would be better.